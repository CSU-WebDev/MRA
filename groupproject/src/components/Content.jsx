import React, {useState} from "react";
import Button from 'react-bootstrap/Button';
import CardList from "./CardList";
import axios from 'axios';
import { useRef } from 'react';


function Content(){
    const [weather, setWeather] = useState()
    const inputRef = useRef(null);
    function fetchdata(){        
        axios.get('https://group-3-backend.onrender.com/?zip='+ inputRef.current.value)
            .then(function (response) {
                setWeather(response.data)
            }).catch(function (error) {
                console.warn("error");
            })
    }
    return(            
        <header className="App-header">  
          <div className="d-flex justify-content-center">              
              <label for="city">Zip Code : </label>
              <input ref={inputRef} type="text" id="city" placeholder="Example: 31907"></input>
              <Button onClick={fetchdata}  variant="warning">Submit</Button>
              
          </div>
          <br />
          <CardList weatherData={weather}/>
        </header>
    )
}

export default Content;