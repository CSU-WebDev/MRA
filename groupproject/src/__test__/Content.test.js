import React from 'react';
import {render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Content from '../components/Content';

test('ChecksTextBoxWithLabel changes after input', () => {
  const {asFragment} = render(<Content />);
        expect(asFragment()).toMatchSnapshot()
});