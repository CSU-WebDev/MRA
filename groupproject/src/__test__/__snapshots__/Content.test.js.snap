// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChecksTextBoxWithLabel changes after input 1`] = `
<DocumentFragment>
  <header
    class="App-header"
  >
    <div
      class="d-flex justify-content-center"
    >
      <label
        for="city"
      >
        Zip Code : 
      </label>
      <input
        id="city"
        name="city"
        placeholder="Example: 31907"
        type="text"
      />
      <button
        class="btn btn-warning"
        type="button"
      >
        Submit
      </button>
    </div>
    <br />
    <div
      class="customCardList scrolls card-group"
    />
  </header>
</DocumentFragment>
`;
